### PR TITLE
feat(video playback): implement picture in picture mode

### PR DESCRIPTION
**Pull request by Camc314**

### CODE REVIEW

File: client/components/Players/PlayerManager.vue [hunk 1]
```diff
@@ -572,6 +572,17 @@ export default Vue.extend({
...
       }
+    },
+    togglePictureInPicture(): void {
+      // @ts-expect-error - `pictureInPictureElement` does not exist in relevant types
+      if (document.pictureInPictureElement) {
+        // @ts-expect-error - `exitPictureInPicture` does not exist in relevant types
+        document.exitPictureInPicture();
+        // @ts-expect-error - `pictureInPictureEnabled` does not exist in relevant types
+      } else if (document.pictureInPictureEnabled) {
+        // @ts-expect-error - `togglePictureInPicture` does not exist in relevant types
```
Comment (dkanada) [inline client/components/Players/PlayerManager.vue:583]: Why not use a different method to disable the errors to avoid four duplicate comments?

Comment (Camc314) [reply client/components/Players/PlayerManager.vue:583]: Such as?

You can't currently comment out blocks in TypeScript @Microsoft/TypeScript#19573

And I couldn't find any type libraries like we have for media sessions

**Comment by Camc314** [comment]
I have removed the `merge conflict` label as it is incorrectly labeled

File: client/components/Players/PlayerManager.vue (hunk 1 above)
Comment (ferferga) [reply client/components/Players/PlayerManager.vue:583]: I think you can create an interface for it with type void

Comment (Camc314) [reply client/components/Players/PlayerManager.vue:589]: I've extended the `document` type to include the relevant functions/expressions. But I'm not a fan of it :/

Comment (dkanada) [reply client/components/Players/PlayerManager.vue:589]: @Camc314 I wasn't aware of the line-specific limitation, I agree the new method is less readable than your previous code :sweat_smile: I think it's fine to revert that change.

Comment (Camc314) [reply client/components/Players/PlayerManager.vue:583]: Ah no worries, I've reverted the change.

File: client/components/Players/PlayerManager.vue [hunk 2]
```diff
@@ -84,7 +84,7 @@
                       <v-btn
                         v-if="supportedFeatures.pictureInPicture"
                         icon
-                        disabled
+                        @click="togglePictureInPicture"
                       >
                         <v-icon> mdi-picture-in-picture-bottom-right </v-icon>
```
Comment (ThibaultNocchi) [inline client/components/Players/PlayerManager.vue:89]: If you could remove the trailing spaces in the meantime :p

**Review by ThibaultNocchi — APPROVED** [review]

File: client/components/Players/PlayerManager.vue (hunk 2 above)
Comment (Camc314) [reply client/components/Players/PlayerManager.vue:89]: Either side of the `v-icon`?

File: client/components/Players/VideoPlayer.vue [hunk 3]
```diff
@@ -247,6 +247,10 @@ export default Vue.extend({
     },
     onPlayerError(event: ErrorEvent): void {
       this.$emit('error', event);
+    },
+    togglePictureInPicture(): void {
+      // @ts-expect-error - `togglePictureInPicture` does not exist in relevant types
```
Comment (MrTimscampi) [inline client/components/Players/VideoPlayer.vue:252]:
```suggestion
      // @ts-expect-error - `requestPictureInPicture` does not exist in relevant types
```

**Review by MrTimscampi — APPROVED** [review]
Small correction in one comment, otherwise it's good :+1: Thanks :smiley:

File: client/components/Players/PlayerManager.vue (hunk 2 above)
Comment (ThibaultNocchi) [inline client/components/Players/PlayerManager.vue:89]:
```suggestion
                        <v-icon>mdi-picture-in-picture-bottom-right</v-icon>
```

Comment (ThibaultNocchi) [reply client/components/Players/PlayerManager.vue:89]: I'm stupid, it's clearer this way ^^

**Review by ThibaultNocchi — APPROVED** [review]

File: client/components/Players/VideoPlayer.vue (hunk 3 above)
Comment (Camc314) [reply client/components/Players/VideoPlayer.vue]: Updated

File: client/components/Players/PlayerManager.vue (hunk 2 above)
Comment (Camc314) [reply client/components/Players/PlayerManager.vue]: Ah no worried. There were a bunch of `v-icons` that had trailing spaces, so I have removed them all in a separate commit.

**Comment by Camc314** [comment]
I have also just wrapped the `player-manager` component in `client-only` tags (from 13a91d5e6656bfcf561aa7ae72f05d45f074b145)

**Review by MrTimscampi — APPROVED** [review]

**Pull request merged**